g1ant-language/desktop-apps-automation.md

```diff
@@ -0,0 +1,306 @@
+# Desktop Apps Automation
+
+One of the most typical examples of robotization on PCs is automating daily tasks, which are done in desktop apps. MS Office applications, for instance, have their own automation engine — Visual Basic macros — to help users change repetitive actions into code in the easiest possible way: by recording their activities in Word or Excel. This recorded actions can then be tweaked and edited in Visual Basic Editor.
+
+It’s a nice feature and lots of people use MS Office macros and VB scripts to ease their everyday chore, but when you want to do something outside of Microsoft’s behemoth suite, say, transfer some data from external sources such as a PDF attachment in an email client, into your Excel spreadsheet, you are left empty-handed.
+
+Well, you would be, if you were forced to rely solely on Microsoft’s solution. Fortunately, G1ANT.Robot is here to the rescue. We will show you how to use it effectively with your common desktop apps, MS Office included, and teach you some G1ANT.Language commands along the way.
+
+## Excel, Procedures and Loops
+
+Let’s start with automating Excel spreadsheets. Imagine you want to get data from one file and enter them into another, newly created file. For the purposes of this tutorial, create a folder called Data in your Documents (Windows 10) or My Documents (earlier Windows versions) folder and put this excel data file [data.xlsx](https://github.com/G1ANT-Robot/G1ANT.Tutorials/blob/master/-assets/data.xlsx) into it.
+
+Since using the whole path to a file every time you want to refer to it is a bit tedious, robotizing this repetitive task would be a good starting point. All you have to do is to declare a new variable — named  `♥datafile1`, for example — with its value equal to the filepath.
+
+The easiest way to find the path to a file is simply to open the file explorer (Windows Explorer), navigate to the folder where your file is stored, and copy the address of the folder from the Explorer’s address bar (to make Explorer display the path in a “classic” way, click folders icon to the left of the bar) followed by the name of the file with extension. The correct filepath should look like this:
+
+C:\Users\\*Username*\Documents\Data\data.xlsx
```

Comment:
double \\ should be single \

Reply:
has to be double, otherwise the "*" will be lost and no italics will appear. If you view the file, this spelling is rendered perfectly.